docs/devel/e2e-tests.md

```diff
@@ -511,6 +512,20 @@ breaking changes, it does *not* block the merge-queue, and thus should run in
 some separate test suites owned by the feature owner(s)
 (see [Continuous Integration](#continuous-integration) below).
 
+### Viper configuration and hierarchichal test parameters.
+
+The future of e2e test configuration idioms will be increasingly defined using viper, and decreasingly via flags.
+
+Flags in general fall apart once tests become sufficiently complicated.  So, even if we could use another flag library, it wouldnt be ideal.
+
+To use viper, rather than flags, to configure youre tests:
+
+- Just add "e2e.json" to the current directory you are in, and define parameters in it... i.e. `"kubeconfig":"/tmp/x"`.
```

Comment:
yeah, we could integrate dumping into the binary at some point.  maybe a follow on PR for that as it would be a separate body of work.